### PR TITLE
Allow Super Admins to do things in organizations that they are not a member of. Also, some cleanup.

### PR DESCRIPTION
**Pull request by NullVoxPopuli**

### CODE REVIEW

File: source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-role.tsx [hunk 1]
```diff
@@ -205,7 +205,7 @@ export function withRole<TWrappedProps extends {}>(
   };
 }
 
-function canDoEverything(dataStore: Store, currentUser: UserResource) {
+export function canDoEverything(dataStore: Store, currentUser: UserResource) {
```
Comment (dalinarkholin) [inline source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-role.tsx]: This feels somewhat like a wrapper of `isSuperAdmin` Looks like the only difference is you are passing currentUser vs the userRoles. Reads well though `const isSuperAdmin = currentUser && canDoEverything(dataStore, currentUser);`

So no action needed. Was thinking maybe changing name to `isUserSuperAdmin` or something...but it's synonymous.

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/components/sidebar/header.tsx [hunk 2]
```diff
@@ -25,90 +29,80 @@ export interface IProps {
...
+  const hasMoreThanOneOrg = organizationsAvailableToUser.length > 1;
+
+  useEffect(() => {
+    if (!currentOrganizationId && organizationsAvailableToUser.length === 1) {
+      const id = idFromRecordIdentity(organizationsAvailableToUser[0] as any);
+
+      setCurrentOrganizationId(id, { skipNav: true });
     }
-  }
+  }, [currentOrganizationId]);
```
Comment (dalinarkholin) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/components/sidebar/header.tsx]: woo hooks 🥇

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/components/sidebar/org-switcher/with-data.tsx [hunk 3]
```diff
@@ -1,81 +1,56 @@
...
-import { defaultOptions, withLoader, attributesFor } from '@data';
+import { defaultOptions, attributesFor } from '@data';
 
-import { IProvidedProps as IFilterProps, withFiltering } from '@data/containers/api/with-filtering';
 import { TYPE_NAME as ORGANIZATION, OrganizationResource } from '@data/models/organization';
-import { ICurrentUserProps, withCurrentUserContext } from '@data/containers/with-current-user';
-import { debounce } from '@lib/debounce';
-import { withRelationships } from '@data/containers/with-relationship';
+import { useCurrentUser } from '@data/containers/with-current-user';
+import _ from 'lodash';
```
Comment (dalinarkholin) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/components/sidebar/org-switcher/with-data.tsx]: probably blind, but is this used?

Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/components/sidebar/org-switcher/with-data.tsx]: nope!

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/components/sidebar/header.tsx (hunk 2 above)
Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/components/sidebar/header.tsx]: woo!
Turns out just by re-writing some things in the new format, I can also get rid of a bunch of previous hacks

File: source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-role.tsx (hunk 1 above)
Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/data/containers/with-role.tsx]: good call tho

**Pull request merged**